ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -475,4 +445,21 @@ private long getWaitingTimeMs() {
       }
     }
   }
+
+  public interface Listener {
+    /**
+     * This method will be called whenever the underlying application
+     * throws an uncaught exception.
+     *
+     * @param error the error that occurred
+     */
+    void onError(QueryMetadata queryMetadata, QueryError error);
+
+    void onStateChange(
+        QueryMetadata queryMetadata,
+        KafkaStreams.State var1,
+        KafkaStreams.State var2);
```

Comment:
ah yeah this is auto-generated. I think the only case we're calling it explicitly is when restarting the query so that the callback gets poked with the new state. In that case we don't know the old value so we're just passing the current value for both.